code/modules/research/xenoarchaeology/artifact/effect.dm

```diff
@@ -23,11 +23,11 @@
 //6 = Interdimensional/bluespace? phasing
 //7 = Atomic synthesis
 
-/datum/artifact_effect/New(var/atom/location)
+//send 1 after location to generate a trigger for the effect, only do this on objects that have the required events!
+/datum/artifact_effect/New(var/atom/location, var/generate_trigger = 0)
 	..()
 	holder = location
 	effect = rand(0,MAX_EFFECT)
```

Comment:
Because we can't have nice things and I'm shite at this whole well designed system thing. Feel free to replace it.